state_version.go

```diff
@@ -55,12 +55,18 @@ type StateVersionList struct {
 
 // StateVersion represents a Terraform Enterprise state version.
 type StateVersion struct {
-	ID           string    `jsonapi:"primary,state-versions"`
-	CreatedAt    time.Time `jsonapi:"attr,created-at,iso8601"`
-	DownloadURL  string    `jsonapi:"attr,hosted-state-download-url"`
-	Serial       int64     `jsonapi:"attr,serial"`
-	VCSCommitSHA string    `jsonapi:"attr,vcs-commit-sha"`
-	VCSCommitURL string    `jsonapi:"attr,vcs-commit-url"`
+	ID                 string                   `jsonapi:"primary,state-versions"`
+	CreatedAt          time.Time                `jsonapi:"attr,created-at,iso8601"`
+	DownloadURL        string                   `jsonapi:"attr,hosted-state-download-url"`
+	Serial             int64                    `jsonapi:"attr,serial"`
+	VCSCommitSHA       string                   `jsonapi:"attr,vcs-commit-sha"`
+	VCSCommitURL       string                   `jsonapi:"attr,vcs-commit-url"`
+	ResourcesProcessed bool                     `jsonapi:"attr,resources-processed"`
+	StateVersion       int                      `jsonapi:"attr,state-version"`
+	TerraformVersion   string                   `jsonapi:"attr,terraform-version"`
+	Modules            *StateVersionModules     `jsonapi:"attr,modules"`
+	Providers          *StateVersionProviders   `jsonapi:"attr,providers"`
+	Resources          []*StateVersionResources `jsonapi:"attr,resources"`
```

Comment:
Can these be tested?

Reply:
@brandonc I can confirm the new attributes are exposed in the response body. Is that sufficient? Do you have any suggestions.

```
{
   "data": {
      "id": "sv-DYXpEz7SQBrRJEHV",
      "type": "state-versions",
      "attributes": {
         "created-at": "2022-07-06T17:54:23.254Z",
         "size": 699,
         "hosted-state-download-url": "https://tfcdev-2c13224a.ngrok.io/_archivist/v1/object/dmF1bHQ6djE6MkpXVGFwQmhqWnRheGdGZEpjUVV4UGFCRFhyVm9nTTV3a044SVF3eldHTGtFVHNpSXpYOVRBWHRNNkU0SVlxMVVObjZPajRId0drUGlXN0N4dTdxRmpTeGV6Yk56SmtIUHRCbVJsN3pGNGhpSWswMnNJUWJ4RDhMcEdnMERwZk5ycFRpemhaVWlyVmZXWmZsYkJxYnVQU0tpOGJrOTYvL21NOWswU1ZQRXkxeWF2WTBxTElJK0gzTlBYYmRpbThsUlpxeDZldUVJQ1FYMEN3VHJXS1d4cmVqRzMrT2xHWVl4NktYWlNpTStTWWhzZ2lDYXExMGZLSjN4ekxJVWNYdjVwaDJtM1hCYWlDUEUzNGtaV0luSDZwbTVzZmcxNG5ZQWVQVkJHZzhUWjFBQk1KRFFrd3BDY0dya3ZjRlM4UktITmpKNjlEMG01NXI1QTJtT3dtd3JCMFlwek50bzY5bXkrMnk1V1FPMW9TM2t1c3BITFpocmNMaFNlOTJ4VjVOVmlhSTVKbnVjdHlNTERaU0gxbkRpMWpCTy9XSlNYQT0",
         "modules": {
            "root": {
               "random_pet": 3
            }
         },
         "providers": {
            "provider[\"registry.terraform.io/hashicorp/random\"]": {
               "random_pet": 3
            }
         },
         "resources": [
            {
               "name": "animal_trio",
               "type": "random_pet",
               "count": 3,
               "module": "root",
               "provider": "provider[\"registry.terraform.io/hashicorp/random\"]"
            }
         ],
         "resources-processed": true,
         "serial": 0,
         "state-version": 4,
         "terraform-version": "1.2.4",
         "vcs-commit-url": null,
         "vcs-commit-sha": null
      },
      "relationships": {
         "run": {
            "data": {
               "id": "run-T3EUTgV3yzSR91GE",
               "type": "runs"
            }
         },
         "next-state-version": {
            "data": null
         },
         "previous-state-version": {
            "data": null
         },
         "created-by": {
            "data": {
               "id": "user-wJVRjZegtRxpexoa",
               "type": "users"
            },
            "links": {
               "self": "/api/v2/users/user-wJVRjZegtRxpexoa",
               "related": "/api/v2/runs/run-T3EUTgV3yzSR91GE/created-by"
            }
         },
         "workspace": {
            "data": {
               "id": "ws-LRczoXxb9a3XSmLh",
               "type": "workspaces"
            }
         },
         "outputs": {
            "data": [
               {
                  "id": "wsout-cJjt3xxikeKDGA1g",
                  "type": "state-version-outputs"
               }
            ],
            "links": {
               "related": "/api/v2/state-versions/sv-DYXpEz7SQBrRJEHV/outputs"
            }
         }
      },
      "links": {
         "self": "/api/v2/state-versions/sv-DYXpEz7SQBrRJEHV"
      }
   }
}
```